torch/_inductor/kernel/mm.py

```diff
@@ -277,23 +277,37 @@ def _is_sm7x_or_older_gpu(index: Optional[int]) -> bool:
 
 def tuned_mixed_mm(mat1, mat2, mat2_dtype):
     m, n, k, layout, mat1, mat2 = mm_args(mat1, mat2, layout=None)
-    choices = [aten_fallback_mixed_mm.bind((mat1, mat2), layout)]
-    if (
+
+    fallback = aten_fallback_mixed_mm.bind((mat1, mat2), layout)
+
+    choices = [fallback]
+
+    # can't use triton kernel unless one of these is true or if running on v100 (numerical issues)
+    skip_triton = (
         mat1.layout.dtype != torch.float32 and not mat2.layout.is_contiguous()
-    ) or _is_sm7x_or_older_gpu(layout.device.index):
-        # can't use triton kernel unless one of these is true or if running on v100 (numerical issues)
-        return autotune_select_algorithm("mixed_mm", choices, [mat1, mat2], layout)
+    ) or _is_sm7x_or_older_gpu(layout.device.index)
+
     if inductor_config.force_mixed_mm:
```

Comment:
In the previous version, this line would never run if the condition that's now skip_triton is True due to the return statement above. If skip_triton is True and the cutlass backend is not enabled or selectable for these inputs, this will ensure that no ATen fallback Kernel is available and autotune_select_algorithm will have nothing to choose from. So, this needs to be adapted a bit.

Reply:
@alexsamardzic - Could you kindly resolve this?

Thank you for your review @kadeng

Reply:
Re-examined this, at the bottom of the method there is:
```
    if skip_triton and not choices:
        choices = [fallback]
```
that I've put there exactly because of the situation you described, so I believe ATen fallback will be selected in this case.